### DISCUSSION TITLE
cm3-ide issue

### DISCUSSION THREAD

**Discussion by dukester** [modula3/cm3#788] [Q&A]
FYI ...

```
cm3ide
NOTICE:  Unable to locate 'examples' folder.
Recovering user state from /home/dnormandin/proj/CM3_IDE.cfg1
starting TCP service
Scanning Packages: Dec 26 12:43...
scan done: Dec 26 12:43
```
Then I ran firefox/localhost:3800
Got the ide. Then - whammo ...

```
***
*** runtime error:
***    Segmentation violation - possible attempt to dereference NIL
***    pc = 0x59b11e = Length + 0x69 in ../src/text/Text.m3
***

zsh: abort      cm3ide
```

**Comment by ghost** [comment]
Sorry, somebody else will have to chime in here.  I wouldn't try to use anything outside of the command-line compiler.

**Reply by dukester** [reply]
Right on! Not important to me anyway. I just thought that I'd try it for hellery! I prefer CLI tools.

**Comment by RodneyBates** [comment]
I don't think much of anyone here has used m3ide much.  I tried
it out sometime in maybe the last year just to see, and got it
up and doing a few simple things, but nothing serious.

**Reply by dukester** [reply]
I was just curious myself. I know where I am with my CLI tools.

**Comment by VictorMiasnikov** [comment]
> FYI ...
> 
> ```
> cm3ide
> NOTICE:  Unable to locate 'examples' folder.
> Recovering user state from /home/dnormandin/proj/CM3_IDE.cfg1
> starting TCP service
> Scanning Packages: Dec 26 12:43...
> scan done: Dec 26 12:43
> ```
> 
> Then I ran firefox/localhost:3800 Got the ide. Then - whammo ...
> 
> ```
> ***
> *** runtime error:
> ***    Segmentation violation - possible attempt to dereference NIL
> ***    pc = 0x59b11e = Length + 0x69 in ../src/text/Text.m3
> ***
> 
> zsh: abort      cm3ide
> ```


 On 2022-11-16 cm3ide is in good state:
= =
![cm3ide -- Opened file M3AST i3 _](https://user-images.githubusercontent.com/48795548/202153412-d0a5524f-d660-4437-aa5a-c5a0725e9b85.png)

= =
